src/middleware.ts

```diff
@@ -16,11 +22,13 @@ const middleware = async (request: NextRequest) => {
     return NextResponse.redirect(new URL('/', request.url))
   }
 
-  return NextResponse.next()
+  return NextResponse.rewrite(url, {
+    headers
+  })
 }
 
 export const config = {
-  matcher: ['/messagebox', '/mypage', '/post', '/auth/redirect']
+  matcher: ['/messagebox', '/mypage', '/post', '/auth/redirect', '/api/:path*']
```

Comment:
matcher에서 messagebox, mypage, post는 왜 필요한건가요??

Reply:
messagebox, mypage, post는 token이 없는경우 접근 불가능한 페이지라 추가해놓았습니다 !

Reply:
```suggestion
  matcher: ['/messagebox', '/post', '/auth/redirect', '/api/:path*']
```

아 효정님 mypage가 shop으로 변경되고 나의 거래 활동, 다른 사용자의 거래 활동에 둘 다 사용되어서 shop에서 처리되어야할 것 같습니다!

Reply:
[변경 완료](https://github.com/price-offer/offer-fe/pull/177/commits/02296ee6ef31a17491f0dc812161df1f1c1df598) !